### DISCUSSION TITLE
Launching SADL IDE from SADL dev environment

### DISCUSSION THREAD

**Discussion by agabaldon** [SemanticApplicationDesignLanguage/sadl#608] [Q&A] [answered]
I installed Eclipse 2020-12 and imported the SADL project. When I launch the SADL IDE from debug/run as an Eclipse Application, it launches Eclipse 2020-03. How do I change it so that it launches Eclipse 2020-12?

**Comment by tuxji** [comment, marked as answer]
To make your SADL projects launch Eclipse 2020-12 instead of 2020-03, you need to edit SADL's target platform definition file (sadl3/com.ge.research.sadl.parent/com.ge.research.sadl.target/com.ge.research.sadl.target.target), save it, and clean/rebuild the SADL projects.  After you edit that file, Eclipse will complain when some requirements won't work together so you'll know which requirements you have to fix.  You'll have to update Xtext from 2.23.0 to 2.24.0 and you may have to update some other lines too.  Then you'll have to update the installation instructions to say install Eclipse 2020-12 instead of 2020-03 (here are some of these files below):

```text
sadl3/com.ge.research.sadl.parent/com.ge.research.sadl3.documentation/WebSite/sadl3/SadlV3Installation.html
26:  Developers,&nbsp;2020-03 is recommended but other newer versions may work as well.)</li>

sadl3/com.ge.research.sadl.parent/com.ge.research.sadl3.documentation/WebSite/sadl3/DeveloperSetup.html
14:  <li>Eclipse IDE for DSL Developers, Version 2020-03 or later, should be

sadl3/com.ge.research.sadl.parent/com.ge.research.sadl3.documentation/WebSite/sadl3/developer/Sadl3BuildInstructions.html
67:  <li>Install Eclipse 2020-03 by one of the methods below.</li>
72:  Eclipse 2020-03 is recommended but newer versions should work.</li>
83:  <li>Select 2020-03 or later as version</li>
```

You'll also have to update the Xtext version in one of the Maven poms too:

```text
sadl3/com.ge.research.sadl.parent/pom.xml
11:             <xtextVersion>2.23.0</xtextVersion>
```

**Comment by agabaldon** [comment]
Thanks @tuxji  I changed the .target file to use 2020-12 eclipse repos and tried to mvn build. I get an error saying it can't find org.junit.jupiter.api/5.6.0.v20200203-2009. What's the proper way to fix/resolve this kind of error dependency? How do you find the right versions of those dependencies?

**Reply by tuxji** [reply]
You can google "eclipse orbit downloads", drill down into the repository corresponding to 2020-12 which is https://download.eclipse.org/tools/orbit/downloads/drops/R20201130205003/, do a "find in page" search for org.junit.jupiter.api, and look at its link to figure out its new version.  But I would simply replace all the explicit version="..." strings with version="0.0.0" in the target file instead, which works fine as long as there is only one version of each dependency available from each update site.  Then we won't ever have to change the version again for each dependency, only the update site's link itself.

**Comment by agabaldon** [comment]
Changed versions to "0.0.0" and it failed to find a bundle:

[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: com.ge.research.sadl 3.5.0.qualifier
[ERROR]   Missing requirement: com.ge.research.sadl 3.5.0.qualifier requires 'osgi.bundle; org.objectweb.asm [7.0.0,8.0.0)' but it could not be found
[ERROR] 
[ERROR] See http://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.

I looked at the orbit downloads repo and for org.objectweb.asm there are no versions within [7.0.0,8.0.0) . I haven't been able to find where that version range requirement comes from.

**Reply by tuxji** [reply]
I found two places where that version range requirement comes from:

```text
sadl3/com.ge.research.sadl.parent/com.ge.research.sadl.tests/META-INF/MANIFEST.MF
11: org.objectweb.asm;bundle-version="[7.0.0,8.0.0)";resolution:=optional,

sadl3/com.ge.research.sadl.parent/com.ge.research.sadl/META-INF/MANIFEST.MF
24: org.objectweb.asm;bundle-version="[7.0.0,8.0.0)";resolution:=optional
```

Instead of updating the version range, let's try removing it.  Change each line above to:

org.objectweb.asm;resolution:=optional
